### ISSUE TITLE
Aggregate field values are not escaped when used in search queries

### ISSUE THREAD

**Issue by TobiasKjrsgaard** [cvtienhoven/graylog-plugin-aggregates#44]
### Background
I've set up an aggregate rule that triggers an alert when there are more than five 404/Not Found exceptions on the same URL within a short time span, using this plugin with the following configuration:

```
backlog: 1
description: The same value of field 'http_raw_url' occurs 5 or more times in a 1 minute interval
field: http_raw_url
grace: 0
interval: 1
match_more_or_equal: true
number_of_matches: 5
query: http_status_code:404 AND NOT request_method:HEAD
repeat_notifications: false
rule_name: Multiple 404's
threshold: 5
threshold_type: MORE_OR_EQUAL
time: 1
type: Aggregates Alert
```

I thought this would enable me to recieve an email notification with the URL whenever this condition was met, but it seems that the alert is never triggered.

### The Issue
After poking around a bit (and raising Graylog's internal log level from INFO to TRACE), I was able to determine that the rule _is_ triggered just fine. The problem is that when the alert is triggered Graylog performs a search to find all messages that match the aggregate field value (`http_raw_url` in my case).

Here's a snippet of Graylog's internal log detailing the issue:

```
2018-10-04T13:14:23.307+02:00 INFO  [AggregatesAlertCondition] 9 found for http_raw_url=example.com/my-test-url-that-is-not-found
2018-10-04T13:14:23.315+02:00 ERROR [AlertScanner] Skipping alert check <Aggregate rule [Multiple 404's] triggered an alert./38bc9935-bf0f-415e-8e1f-52bbc6630274>
org.graylog2.indexer.ElasticsearchException: Unable to perform search query

Failed to parse query [http_status_code:404 AND NOT request_method:HEAD AND http_raw_url: example.com/my-test-url-that-is-not-found]
	at org.graylog2.indexer.cluster.jest.JestUtils.specificException(JestUtils.java:98) ~[graylog.jar:?]
	at org.graylog2.indexer.searches.Searches.wrapInMultiSearch(Searches.java:957) ~[graylog.jar:?]
	at org.graylog2.indexer.searches.Searches.search(Searches.java:284) ~[graylog.jar:?]
	at org.graylog2.indexer.searches.Searches.search(Searches.java:268) ~[graylog.jar:?]
	at org.graylog.plugins.aggregates.alerts.AggregatesAlertCondition.runCheck(AggregatesAlertCondition.java:151) ~[?:?]
	at org.graylog.plugins.aggregates.alerts.AggregatesAlertCondition.runCheck(AggregatesAlertCondition.java:42) ~[?:?]
	at org.graylog2.alerts.AlertScanner.checkAlertCondition(AlertScanner.java:64) ~[graylog.jar:?]
	at org.graylog2.periodical.AlertScannerThread.lambda$doRun$0(AlertScannerThread.java:63) ~[graylog.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1257) [?:1.8.0_181]
	at org.graylog2.periodical.AlertScannerThread.doRun(AlertScannerThread.java:63) [graylog.jar:?]
	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [graylog.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_181]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_181]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
```

It seems that the value of the aggregate field is added to the search query without escaping special characters like the slashes in a URL and without wrapping the value in quotes. So when Graylog tries to perform a search with a URL like "example.com/my-test-url-that-is-not-found", it fails because the slashes aren't escaped which renders the query invalid.

I' not sure if this is an issue with this specific plugin or with Graylog in general?

**Comment by cvtienhoven** [comment]
Hi @TobiasKjrsgaard, thanks for reaching out. This is a bug in the plugin when using the message backlog. It searches the messages for the found values, but doesn't escape these values first. I'm going to try to fix this in the next bugfix release. In the meantime as a workaround you could use the alerts without the message backlog.

**Comment by cvtienhoven** [comment]
Hey @TobiasKjrsgaard, I just released version 2.4.0 of the plugin, which escapes the search query when assembling the message backlog. Could you give this a test? You will need to run a 2.5.x version of Graylog though, the latest version of the plugin is not compatible with 2.4.x anymore.

**Comment by TobiasKjrsgaard** [comment]
I just tried the latest plugin version on Graylog 2.5.1; it works flawlessly now.

Thanks!
